napari/utils/colormaps/colormap.py

```diff
@@ -81,3 +85,78 @@ def map(self, values):
     @property
     def colorbar(self):
         return make_colorbar(self)
+
+
+@evented_dataclass
+class CategoricalColormap:
```

Comment:
> I might put this in it's own file categorical_colormap.py.

I am fine with this.

>  Does it also need a colorbar?

Hmm good question. I think we should have some sort of visual representation. I'd like to punt on this until we update the layer controls GUI for the `ColorManager` so that we can design all of GUI stuff together.